Library/Homebrew/download_strategy.rb

```diff
@@ -222,27 +222,30 @@ def extract_ref(specs)
 
 class AbstractFileDownloadStrategy < AbstractDownloadStrategy
   def stage
-    case cached_location.compression_type
+    case type = cached_location.compression_type
     when :zip
       with_system_path { quiet_safe_system "unzip", "-qq", cached_location }
       chdir
     when :gzip_only
       with_system_path { buffered_write("gunzip") }
     when :bzip2_only
       with_system_path { buffered_write("bunzip2") }
-    when :gzip, :bzip2, :compress, :tar
+    when :gzip, :bzip2, :xz, :compress, :tar
       tar_flags = "x"
       # Older versions of tar require an explicit format flag
-      if cached_location.compression_type == :gzip
+      if type == :gzip
         tar_flags << "z"
-      elsif cached_location.compression_type == :bzip2
+      elsif type == :bzip2
         tar_flags << "j"
       end
       tar_flags << "f"
-      with_system_path { safe_system "tar", tar_flags, cached_location }
-      chdir
-    when :xz
-      with_system_path { pipe_to_tar(xzpath) }
+      with_system_path do
+        if type == :xz && MacOS.version < :lion
+          pipe_to_tar(xzpath)
+        else
+          safe_system "tar", tar_flags, cached_location
```

Comment:
You're proposing making a change to additional code so the onus is not on me to research why I shouldn't change the existing code. It's not a bug we're seeing and I don't think speculative fixing like that is a sensible approach as it may break things for forks like Linuxbrew and it's not gaining us anything today. Again, happy to have a longer-term PATH-filtering solution that addresses all these cases and avoids us having to hard-code anything.